snooty/parser.py

```diff
@@ -700,138 +705,150 @@ def create_page() -> Tuple[Page, EmbeddedRstParser]:
+                    self.diagnostics.append(ExpectedPathArg(name, line))
+                    return doc
+            if argument_text is not None:
+                _, filepath = util.reroot_path(
+                    FileId(argument_text), self.docpath, self.project_config.source_path
                 )
-                return doc
-            except UnicodeDecodeError as err:
-                self.diagnostics.append(CannotOpenFile(argument_text, str(err), line))
-                return doc
 
-            lines = text.split("\n")
-
-            def _locate_text(text: str) -> int:
-                """
-                Searches the literally-included file ('lines') for the specified text. If no such text is found,
-                add an InvalidLiteralInclude diagnostic.
-                """
-                assert isinstance(text, str)
-                loc = next((idx for idx, line in enumerate(lines) if text in line), -1)
-                if loc < 0:
+                # Attempt to read the literally included file
+                try:
+                    text = filepath.read_text(encoding="utf-8")
+                except OSError as err:
                     self.diagnostics.append(
-                        InvalidLiteralInclude(f'"{text}" not found in {filepath}', line)
+                        CannotOpenFile(argument_text, err.strerror, line)
                     )
-                return loc
-
-            # Locate the start_after query
-            start_after = 0
-            if "start-after" in options:
-                start_after_text = options["start-after"]
-                start_after = _locate_text(start_after_text)
-                # Only increment start_after if text is specified, to avoid capturing the start_after_text
-                start_after += 1
-
-            # ...now locate the end_before query
-            end_before = len(lines)
-            if "end-before" in options:
-                end_before_text = options["end-before"]
-                end_before = _locate_text(end_before_text)
-
-            # Check that start_after_text precedes end_before_text (and end_before exists)
-            if start_after >= end_before >= 0:
-                self.diagnostics.append(
-                    InvalidLiteralInclude(
-                        f'"{end_before_text}" precedes "{start_after_text}" in {filepath}',
-                        line,
+                    return doc
+                except UnicodeDecodeError as err:
+                    self.diagnostics.append(
+                        CannotOpenFile(argument_text, str(err), line)
                     )
-                )
+                    return doc
 
-            # If we failed to locate end_before text, default to the end-of-file
-            if end_before == -1:
-                end_before = len(lines)
+                lines = text.split("\n")
+                # Capture the original file-length before splicing it
+                len_file = len(lines)
```

Comment:
this can be defined closer to where it's used down below

Reply:
Actually since lines gets changed within the next if block, and len_file depends on the original file-length before it gets altered, I think it needs to stay there!